electrum/gui/qml/qeinvoice.py

```diff
@@ -493,79 +490,54 @@ def validateRecipient(self, recipient):
-        except InvalidBitcoinURI as e:
-            bip21 = None
-
-        lninvoice = None
-        maybe_lightning_invoice = maybe_extract_lightning_payment_identifier(maybe_lightning_invoice)
-        if maybe_lightning_invoice is not None:
-            if maybe_lightning_invoice.startswith('lnurl'):
-                self.resolve_lnurl(maybe_lightning_invoice)
-                return
-            try:
-                lninvoice = Invoice.from_bech32(maybe_lightning_invoice)
-            except InvoiceError as e:
-                e2 = e.__cause__
-                if isinstance(e2, LnInvoiceException):
-                    self.validationError.emit('unknown', _("Error parsing Lightning invoice") + f":\n{e2}")
-                    self.clear()
-                    return
-                if isinstance(e2, lnutil.IncompatibleOrInsaneFeatures):
-                    self.validationError.emit('unknown', _("Invoice requires unknown or incompatible Lightning feature") + f":\n{e2!r}")
-                    self.clear()
-                    return
-                self._logger.exception(repr(e))
+        self._update_from_payment_identifier()
 
-        if not lninvoice and not bip21:
-            self.validationError.emit('unknown',_('Unknown invoice'))
-            self.clear()
+    def _update_from_payment_identifier(self):
+        if self._pi.need_resolve():
+            self.resolve_pi()
             return
 
-        if lninvoice:
-            if not self._wallet.wallet.has_lightning():
-                if not bip21:
-                    if lninvoice.get_address():
-                        self.setValidLightningInvoice(lninvoice)
-                        self.validationSuccess.emit()
-                    else:
-                        self.validationError.emit('no_lightning',_('Detected valid Lightning invoice, but Lightning not enabled for wallet and no fallback address found.'))
-                else:
-                    self._logger.debug('flow with LN but not LN enabled AND having bip21 uri')
-                    self._validateRecipient_bip21_onchain(bip21)
-            else:
-                if not self._wallet.wallet.lnworker.channels:
-                    if bip21 and 'address' in bip21:
-                        self._logger.debug('flow where invoice has both LN and onchain, we have LN enabled but no channels')
-                        self._validateRecipient_bip21_onchain(bip21)
-                    else:
-                        self.validationWarning.emit('no_channels',_('Detected valid Lightning invoice, but there are no open channels'))
-                else:
+        if self._pi.type == 'lnurl':
+            self.on_lnurl(self._pi.lnurl_data)
+            return
+
+        if self._pi.type == 'bip70':
+            self._bip70_payment_request_resolved(self._pi.bip70_data)
+            return
+
+        if self._pi.is_available():
+            if self._pi.type == 'spk':
```

Comment:
This looks broken atm for non-address spks -- although it's a question whether we even want to support those in qml, or even if so, whether to do it now.
Still, atm it is broken in a weird unintuitive way.
Try pasting `022782ba871eeb444a202f875c639a5e51bbdd38c5a0610111162cec9a2201254b OP_CHECKSIG`.